pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -534,6 +534,10 @@ func (h *NetworkUtilsHandler) CreateRADaemon(socketPath string, advertisementIfa
 		}
 	}()
 
+	go func() {
```

Comment:
> Can you please explain in the commit message with this periodic RA is needed? Doesn't the client send a new RS when the on RA is about to expire? 

Not that I know of.

> If not, why? 

Do you have any indication why it should ?.. AFAIU, RouterSolicitations are sent when an interface comes up. I am not aware of it being sent on any other type of event - least of all when the routes go missing.

> Maybe it is because the `RouterLifetime` or other lifetime we configure wrong.

Do you have any indication about this ? afaiu, you need to send periodic router advertisements to keep the routes alive.

> 
> Anyway, I believe the `PeriodicallySendRAs` should be implementation detail of the `Serve` and not a separate go routine.

This one is making my brain itch; I don't have a good reply right now, will update once I figure this out (not saying your suggestion does not make sense. Just that I'm not jumping into that band wagon right away).

Reply:
The periodic RAs are waiting on a `select` channel call, whereas replying to RSs is blocked waiting for data using an infinite for loop. 

I can't combine those - unless I change the listening for RSs to notifying a channel.